rfcs/008-map-indexes.md

```diff
@@ -0,0 +1,202 @@
+
+A third option would be to split the document across multiple keys in FoundationDB. This will still be limited by the transaction size limit.
+
+### Index building
+
+An index will be built and updated via a [background job worker](https://github.com/apache/couchdb-documentation/blob/master/rfcs/007-background-jobs.md). When a request for a view is received, the request process will add a job item onto the background queue for the index to be updated. A worker will take the item off the queue and update the index. Once the index has been built, the background job server will notify the request that the index is up to date. The request process will then read from the index and return the results. This process can also be optimised in two ways. Firstly, using a new couch_events system to listen for document changes in a database and then adding indexing jobs to the queue to keep indexes warm. The second optimisation is if the index only requires a small update, rather update the index in the HTTP request process instead of doing the work via the background queue.
+
+Initially, the building of an index will be a single worker running through the changes feed and creating the index. In the future, we plan to parallelise that work so that multiple workers could build the index at the same time. This will reduce build times.
+
+### View clean up
+
+When a design document is changed, new indexes will be built and grouped under a new `View Signature`. The old map indexes will still be in FDB. To clean up will be supported via the existing [/db/_view_cleanup](https://docs.couchdb.org/en/latest/api/database/compact.html#db-view-cleanup) endpoint. 
+
+A future optimisation would be to automate this and have CouchDB to monitor design doc changes and then look to clean up old view indexes via a background worker.
+
+### Stale = “ok” and stable = true
+
+With the consistency guarantee’s CouchDB will get from FDB, `stable = true` will no longer be an option that CouchDB would support and so the argument would be ignored. Similar `stale = “ok”` would now be translated to `update = false`.
+
+### Size limits
+
+- The sum of all keys emitted for a document cannot exceed 100 KB
+- Emitted keys will not be able to exceed 10 KB
+- Values cannot exceed 100 KB
+- There could be rare cases where the number of key-value pairs emitted for a map function could lead to a transaction either exceeding 10 MB in size which isn’t allowed or exceeding 5 MB which impacts the performance of the cluster. In this situation, CouchDB will send an error.
+
+These limits are the hard limits imposed by FoundationDB. We will have to set the user imposed limits to lower than that as we store more information than just the user keys and values.
+
+## Advantages
+
+- Map indexes will work on FoundationDB with the same behaviour as current CouchDB 1.x
+- Options like stale = “ok” and ‘stable = true’ will no longer be needed
```

Comment:
Agreed on `stable=true` but `stale=ok` is still a meaningful option in this design.